src/wp-content/themes/twentytwelve/js/navigation.js

```diff
@@ -25,31 +59,34 @@
 			menu.className = 'nav-menu';
 		}
 
-		if ( -1 !== button.className.indexOf( 'toggled-on' ) ) {
-			button.className = button.className.replace( ' toggled-on', '' );
-			menu.className = menu.className.replace( ' toggled-on', '' );
-		} else {
-			button.className += ' toggled-on';
-			menu.className += ' toggled-on';
-		}
+		toggleClass( button, 'toggled-on' );
+		toggleClass( menu, 'toggled-on' );
 	};
-} )();
 
-// Better focus for hidden submenu items for accessibility.
-( function( $ ) {
-	$( '.main-navigation' ).find( 'a' ).on( 'focus.twentytwelve blur.twentytwelve', function() {
-		$( this ).parents( '.menu-item, .page_item' ).toggleClass( 'focus' );
+	// Better focus for hidden submenu items for accessibility.
+	function toggleParentsFocusClass() {
+		getParentElements( this, '.menu-item, .page_item' ).forEach( function( parentElement ) {
+			toggleClass( parentElement, 'focus' );
+		} );
+	}
+	Array.from( document.querySelector( '.main-navigation' ).getElementsByTagName( 'a' ) ).forEach( function( menuLink ) {
+		menuLink.addEventListener( 'focus', toggleParentsFocusClass );
+		menuLink.addEventListener( 'blur', toggleParentsFocusClass );
 	} );
 
-  if ( 'ontouchstart' in window ) {
-    $('body').on( 'touchstart.twentytwelve',  '.menu-item-has-children > a, .page_item_has_children > a', function( e ) {
-      var el = $( this ).parent( 'li' );
-
-      if ( ! el.hasClass( 'focus' ) ) {
-        e.preventDefault();
-        el.toggleClass( 'focus' );
-        el.siblings( '.focus').removeClass( 'focus' );
-      }
-    } );
-  }
-} )( jQuery );
+	if ( 'ontouchstart' in window ) {
+		Array.from( document.querySelectorAll( '.menu-item-has-children > a, .page_item_has_children > a' ) ).forEach( function( menuLink ) {
+			menuLink.addEventListener( 'touchstart', function( e ) {
+				var el = getParentElements( this, 'li' )[0];
+
+				if ( -1 === el.className.indexOf( 'focus' ) ) {
```

Comment:
```suggestion
				if ( el.classList.contains( 'focus' ) ) {
```

Also, I believe Yoda is only meant to be a thing in PHP code, at least going by [the equality examples in the JS Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#equality).

Reply:
Correct, I think your suggested code needs to be negated though. Will update.